tests/src/nodes/nomos.rs

```diff
@@ -294,11 +292,11 @@ fn create_node_config(
     };
     #[cfg(feature = "waku")]
     {
-        config.network.backend.inner.port = Some(get_available_port() as usize);
```

Comment:
how do nodes client know who to contact if the OS decides the port to use? It's a different process, we can't access the address like you do in mixnet node

Reply:
This was solved by PR #330. In this PR, we can get the correct address through the `MixnodeHandle` struct.

Reply:
I mean for standard nodes (https://github.com/logos-co/nomos-node/blob/b692043a0a45aa94cfdf06e0e1be036e42cda285/tests/src/nodes/nomos.rs#L60), not mixnodes

Reply:
Ah, sorry, I forgot the standard node. Yeah, it is hard to get the port for the standard node, I will rethink it. Thanks!

Reply:
Do we have problems with the current solution?

Reply:
https://github.com/logos-co/nomos-node/pull/315 Even after merging this, I saw 1~2 port conflicts, but I don't remember exactly. Al, did you get the errors after the PR is merged?

Acutally, `tests/mixnet.rs` doens't use `get_available_port()` yet. We should fix it.

Reply:
I do not meet problems with the current solutions for the standard node, so I think we can use `get_avaliable_port` when bootstrapping for the standard node. But, random port assignment is a problem when doing benchmarks for mixnode, we can use auto port for mixnodes.

Reply:
I actually think https://github.com/logos-co/nomos-node/pull/315 made things worse in a way because now you can have conflicts in a single process (previous solution was fine with multiple threads in the same process):
* process 1 ask for some port, receives port 8002, but does not start yet
* process 2 ask for some port, we are unlucky and get port 8002 from the random generator
Since https://github.com/logos-co/nomos-node/pull/315 is not using any global lock and process 1 has not started, the os reports port 8002 as free, but later they will both try to use it

Reply:
Yes. I think that both of previous and current solution have this multiprocess problem, since we don’t have cross-process locking. 

I thought that the probability of conflict between processes is lower in the current solution. In the previous solution, all processes start with the port 800x (I don’t remember the exact number). This will cause the high probability of conflict. 

For single process, you’re right. We may need to revive the lock or something

Reply:
How about reviving the lock from the previous soluton
```
static NET_PORT: Mutex<u16> = Mutex::new(8000);
```
but instead of using the constant 8000, what do you think about using a random start port? 

Using this lock, the single process would be happy. With a random start port, we can lower the probability of conflicts between processes.

Reply:
I'll make a simple PR for this.

Reply:
> I'll make a simple PR for this.

Opened a PR: https://github.com/logos-co/nomos-node/pull/334

>But, random port assignment is a problem when doing benchmarks for mixnode, we can use auto port for mixnodes.

As I discussed with Al, I think it would be better to postpone this PR along with https://github.com/logos-co/nomos-node/pull/328, until we make CI happy for other PRs.